internal/install/execution/install_status.go

```diff
@@ -381,3 +375,65 @@ func (s *InstallStatus) getStatus(r types.Recipe) *RecipeStatus {
 
 	return nil
 }
+
+// This function handles updating the final recipe statuses and top-level installation status.
+// Canceling (e.g. ctl+c) will cause unresolved recipes to be marked as canceled.
+// Exiting early (i.e. an error occurred) will cause unresolved recipes to be marked as failed.
+func (s *InstallStatus) updateFinalInstallationStatuses(installCanceled bool) {
```

Comment:
Self review: Needing to pass a boolean (or some indicator/value) to differentiate between canceled/complete to this method presented some code smell. I tried to DRY up some of the repeated code in the `for` loop. Doing so left me thinking we might want to consider consolidating `InstallStatus.InstallCanceled()` and `InstallStatus.InstallComplete()` into a single method that captures the idea of "install exited" and then handle setting "failed" vs "canceled" in there, but even that seems like we can make setting the installation and recipe states in a cleaner way in a future PR.